pkg/broker/ingress/multi_topic_decouple_sink.go

```diff
@@ -90,6 +97,26 @@ func (m *multiTopicDecoupleSink) Send(ctx context.Context, broker types.Namespac
 	return err
 }
 
+// EventFilterFunc is used to see if a target is interested in an event.
+// It is used as a vaiable to allow stubbing out in unit tests.
+var EventFilterFunc = filter.PassFilter
+
+// hasTrigger checks given event against all targets to see if it will pass any of their filters.
+// If one is fouund, hasTrigger returns true.
+func (m *multiTopicDecoupleSink) hasTrigger(ctx context.Context, event *cev2.Event) bool {
+	var hasTrigger bool
+	m.brokerConfig.RangeAllTargets(func(target *config.Target) bool {
```

Comment:
I think we should gate this feature behind a flag and disable it by default. Alternatively we can make sure we do some performance testing before we release. However, the latter could potentially delay our release.

Thoughts? @grantr

Reply:
IMO we don't need a flag for this, and we should test the performance impact anyway.

If we're concerned about filtering taking too long in edge cases, we can add a timeout after which the event is sent to the decoupling topic (and a prefilter timeout metric is incremented). This would be similar to fanout's initial delivery timeout to retry (but with a much shorter timeout, maybe 100ms).

Reply:
As discussed, I added a feature flag for this until we test the performance change. I've added TODOs to clean up the feature gate as part of #1804.